frontend/packages/console-shared/src/components/dashboard/utilization-card/UtilizationItem.tsx

```diff
@@ -68,4 +92,8 @@ type UtilizationItemProps = {
   query: string;
   error: boolean;
   max?: number;
+  PopupComponent?: React.ComponentType<any>;
```

Comment:
it would be better for PopupComponent to have all props already so instead of

```javascript
{PopupComponent && !isLoading ? (
              <DashboardCardPopupLink popupTitle={`${title} breakdown`} linkTitle={current}>
                <PopupComponent
                  humanize={humanizePopup}
                  topConsumers={topConsumers}
                  title={title}
                  query={popupQuery}
                />
              </DashboardCardPopupLink>
            ) : (
              current
            )}

```
you would do
```
{PopupComponent && !isLoading ? <PopupComponent linkTitle={current} /> : current}
```

then there's no need to pass `topConsumers`, `humanizePopup` and `popupQuery` to `UtilizationItem`

Reply:
How would I initialize PopupComponent with the required props and pass it as a prop? Is it possible to do that without passing the popup component as a child?

Reply:
In Utilization card you could do 

```javascript
const CPUPopover = React.useCallback(
    ({ current }) => (
      <DashboardCardPopupLink popupTitle="CPU breakdown" linkTitle={current}>
        <ConsumerPopover query={cpuQueriesPopup} humanize={humanizeSeconds} ...any other props/>
      </DashboardCardPopupLink>
    ),
    [],
  );
```

and pass that to UtilizationItem
```javascript
<UtilizationItem
  title="CPU"
  data={cpuStats}
  error={cpuUtilizationError}
  isLoading={!cpuUtilization}
  humanizeValue={humanizePercentage}
  query={queries[OverviewQuery.CPU_UTILIZATION]}
  PopupComponent={CPUPopover}
/>
```

and in UtilizationItem you would have

```javascript
<div className="pf-l-level">
  <h4 className="pf-c-title pf-m-md">{title}</h4>
  {PopupComponent && !error ? <PopupComponent current={current} /> : current}
</div>
```